Gp 7982/update deployment config ticket processing

ryanfreckleton opened the pull request:
Initial attempt to break Helm chart into two versions, since we'll want to deploy the new Ticket Workers for testing while maintaining the old version for production and patch releases.

Commit:
GP-7982: Create first draft of ticket worker.

This just copies the AOD Worker configuration. The details of the config
values will be added/modified once that lands in Data-Pipeline.

https://tickets.gitprime-ops.com/browse/GP-7982

Commit:
NO-TICKET: Add README

Commit:
GP-7982: Move into separate version directory

toddjtidwell commented on charts/gitprime-app/v2.0.0/Chart.yaml:
```diff
@@ -0,0 +1,6 @@
+apiVersion: v1
+appVersion: "1.0"
```
Might as well change this to 2.0 as well

ryanfreckleton replied:
:sweat_smile:

toddjtidwell commented:
I think you're going to want to rename the "latest" directory to 1.0.0.  That way we have a directory per version, and that directory name won't be confusing.  I *THINK* there is even a way to do a "link" from "latest" to the correct version we're treating as the latest dev version.

Commit:
GP-7982: Set appVersion to 2.0

Commit:
GP-7982: Rename 'latest' folder to 'v1.0.0'

ryanfreckleton commented:
Renamed "latest" to "v1.0.0". I couldn't find a way to 'link' a "latest" folder to the appropriate version after about 20 minutes of googling.

dcumberland commented on charts/gitprime-app/v2.0.0/questions.yml:
```diff
@@ -0,0 +1,451 @@
+    description: The maximum amount of memory a new repository worker can use.
+    default: "8Gi"
+
+  - variable: dataPipeline.newRepoWorker.requeueAOD
```
RequeueAOD is no longer needed.  I believe this next version of the app "embeds" this decision.  That is due to, in part, a separation of the commit processing "message consumer" and a new AOD "message consumer".

dcumberland commented on charts/gitprime-app/v2.0.0/templates/dataPipeline/_deployment.tpl:
```diff
@@ -0,0 +1,114 @@
+              value: {{ quote .templateData.operationMode }}
+            - name: SYSTEM_CONFIG_SERVER_URL
+              value: {{ include "helpers.configServerURL" . }}
+            - name: GP_REQUEUE_AOD
```
Ripple effect of removing the AOD requeue config, this env variable can be removed and we should verify that the config jsons no longer look for it.

dcumberland commented on charts/gitprime-app/v2.0.0/templates/dataPipeline/_deployment.tpl:
```diff
@@ -0,0 +1,114 @@
+          {{- end }}
+          {{- if .templateData.maxConcurrentJobs }}
+            - name: GP_MAX_CONCURRENT_JOBS
+              value: {{ quote .templateData.maxConcurrentJobs }}
```
The config.json will now have separate values for each queue type, each value.  So things like `GP_MAX_CONCURRENT_JOBS` need to be given with the queue type.

In practice, we will only run one queue type per pod in production.  However, we want to be able to run multiple queues in one pod in other environments, which means we will need to separate out each of these per-queue configs.

dcumberland reviewed (changes requested):
There are implications of being able to run multiple queues in one JVM that need to get worked out in the helm chart.  Overall, it should simplify what the helm chart has to do, but it does mean possibly reworking structure.

Commit:
GP-7982: Add env variables for new modes.

Each worker type is controlled by environment variables that set the
number of threads that listen to each queue.

Commit:
GP-7982: Remove requeueAOD.

dcumberland reviewed (approved):
LGTM
